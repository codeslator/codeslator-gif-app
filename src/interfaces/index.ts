import { Dispatch, SetStateAction } from 'react'

export interface AddCategoryProps {
  categories: string[],
  setCategories: Dispatch<SetStateAction<string[]>>
}

export type GifGridProps = {
  category: string;
}