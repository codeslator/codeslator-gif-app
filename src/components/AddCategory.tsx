import { useState, FormEvent, SyntheticEvent, FC } from 'react';
import { AddCategoryProps } from '../interfaces/index';

export const AddCategory: FC<AddCategoryProps> = ({ categories, setCategories }) => {
  const [value, setValue] = useState<string>('');

  const handleInput = (event: FormEvent<HTMLInputElement>) => {
    setValue(event.currentTarget.value)
  };

  const handleSubmit = (e: SyntheticEvent) => {
    e.preventDefault();
    if(value !== '') {
      setCategories((prev: string[]) => [...prev, value]);
      setValue('');
    }
  };

  return (
    <>
      <form onSubmit={handleSubmit}>
        <h2>Search Category</h2>
        <div className="row">
          <div className="col-9">
            <input
              type="text"
              className="form-control"
              placeholder="Category"
              id="category"
              value={value}
              onChange={handleInput}
            />
          </div>
          <div className="col-3">
            <button className="btn btn-primary btn-full" type="submit">Submit</button>
          </div>
        </div>
      </form>
      <p>{value}</p>
    </>
  )
}
