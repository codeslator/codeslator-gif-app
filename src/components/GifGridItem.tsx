import { GifGridItemProps } from '../interfaces/index';

export const GifGridItem = ({ gif }: GifGridItemProps) => {
  const { title, url } = gif;
  return (
    <div className="card mb-3 animate__animated animate__fadeIn">
      <img src={url} className="card-img-top" alt={title} />
      <div className="card-body">
        <h5 className="card-title">{title}</h5>
      </div>
    </div>
  )
}
