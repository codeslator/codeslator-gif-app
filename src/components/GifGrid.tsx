import { FC, useEffect } from 'react';
import { GifGridProps } from '../interfaces/index';
import useGif from '../hooks/useGif';
import { GifGridItem } from './GifGridItem';

export const GifGrid: FC<GifGridProps> = ({ category }) => {
  const { fetchGifs, gifs, isLoading } = useGif();

  useEffect(() => {
    fetchGifs(category)
  }, [category, fetchGifs])

  return (
    <>
      <h3 className="mb-3">{category}</h3>
      <div className="row">
        {isLoading && (
          <h4 className="text-center">Loading...</h4>
        )}
        {gifs.map((gif) => (
          <div className="col-xs-6 col-sm-4 col-md-3 col-lg-2" key={gif.id} >
            <GifGridItem gif={gif} />
          </div>
        ))}
      </div>
    </>
  )
}
