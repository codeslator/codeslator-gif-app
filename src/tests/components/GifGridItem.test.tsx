import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';
import { Gif } from '../../interfaces';

describe('Testing GifGridItem component', () => {
  const gif: Gif = {
    id: '1',
    title: 'A title',
    url: 'https://localhost:3000/image.png'
  };
  const wrapper = shallow(<GifGridItem gif={gif} />);


  test('Should to show the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should to have a h5 as title', () => {
    const h5 = wrapper.find('h5');
    expect(h5.text().trim()).toBe(gif.title);
  });

  test('Image should have the props equals as url and alt', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(gif.url);
    expect(img.prop('alt')).toBe(gif.title);
  });

  test('Should includes animate__fadeIn class', () => {
    const div = wrapper.find('div.card');
    const className = div.prop('className')
    expect(className?.includes('animate__fadeIn')).toBeTruthy();
  })
})