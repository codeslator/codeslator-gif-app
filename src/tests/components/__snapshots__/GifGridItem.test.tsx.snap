// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem component Should to show the component correctly 1`] = `
<div
  className="card mb-3 animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    className="card-img-top"
    src="https://localhost:3000/image.png"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      A title
    </h5>
  </div>
</div>
`;
